Subject: [PATCH] modify ua
---
Index: transport/internet/grpc/dial.go
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/transport/internet/grpc/dial.go b/transport/internet/grpc/dial.go
--- a/transport/internet/grpc/dial.go	(revision 28f558601fb8b6ab1bf4cdbdd3b8848047393e77)
+++ b/transport/internet/grpc/dial.go	(revision 294e80a51a8839ed8ec01d6caf2152187e8cbf12)
@@ -24,6 +24,14 @@
 	"github.com/v2fly/v2ray-core/v5/transport/internet/tls"
 )

+type withUAOption struct {
+	ua string
+}
+
+func (o *withUAOption) apply(dOpts *grpc.dialOptions) {
+	dOpts.copts.UserAgent = o.ua
+}
+
 func Dial(ctx context.Context, dest net.Destination, streamSettings *internet.MemoryStreamConfig) (internet.Connection, error) {
 	newError("creating connection to ", dest).WriteToLog(session.ExportIDToError(ctx))

@@ -87,6 +95,7 @@
 	if client, found := globalDialerMap[dest]; found && client.GetState() != connectivity.Shutdown {
 		return client, canceller, nil
 	}
+	uaOpt := &withUAOption{ua: "gRPC.xClient.2025"}

 	conn, err := grpc.Dial(
 		dest.Address.String()+":"+dest.Port.String(),
@@ -116,6 +125,7 @@
 			detachedContext := core.ToBackgroundDetachedContext(ctx)
 			return internet.DialSystem(detachedContext, net.TCPDestination(address, port), streamSettings.SocketSettings)
 		}),
+		uaOpt,
 	)
 	globalDialerMap[dest] = conn
 	return conn, canceller, err
